docs/tutorials/basics/50_control_flow.md

```diff
@@ -240,3 +240,130 @@ end
 
 Both clauses have branches with the same conditions but in a different order and they behave differently.
 The first matching condition selects which branch executes.
+
+## Loops
+
+This section delves into reducing repetition by letting code repeat itself in a loop.
+
+The basic feature is the `while` clause. Its structured quite similar to an `if` clause:
+The keyword `while` designates the begin and is followed by an expression serving as the loop condition.
+All subsequent expressions are part of the loop until the closing keyword `end`.
+The loop continues to repeat itself as long as the return value of the condition is *truthy*.
+
+Let's try a simple program for counting from 1 to 10:
+
+```{.crystal .crystal-play}
+counter = 0
+
+while counter < 10
+  counter += 1
+
+  puts "Counter: #{counter}"
+end
+```
+
+The code between `while` and `end` is executed 10 times. It prints the current counter value and increases it by one.
+After the 10th iteration, the value of `counter` is `10`, thus `counter < 10` fails and the loop breaks.
+
+An alternative is to replace `while` with the keyword `until` which expects just the opposite truthiness. `until x` is equivalent to `while !x`.
+
+```crystal-play
+counter = 0
+
+until counter > 9
+  counter += 1
+
+  puts "Counter: #{counter}"
+end
+```
+
+!!! tip
+    You can find more details on these expressions in the language specification: [`while`](../../syntax_and_semantics/while.md) and [`until`](../../syntax_and_semantics/until.md).
+
+### Infinite loops
+
+When working with loops, it's important to care about the loop condition failing at some point.
+Otherwise, it wouldn't stop ever and continue to all eternity – or until you stop the program externally (for example <kbd>Ctrl+C</kbd>, `kill`, pull the plug or when armageddon arrives).
```

Comment:
```suggestion
Otherwise, it would continue forever or until you stop the program externally (for example <kbd>Ctrl+C</kbd>, `kill`, pull the plug or when armageddon arrives).
```